chia/wallet/conditions.py

```diff
@@ -0,0 +1,1221 @@
+    ConditionOpcode.ASSERT_SECONDS_RELATIVE.value: AssertSecondsRelative,
+    ConditionOpcode.ASSERT_SECONDS_ABSOLUTE.value: AssertSecondsAbsolute,
+    ConditionOpcode.ASSERT_HEIGHT_RELATIVE.value: AssertHeightRelative,
+    ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE.value: AssertHeightAbsolute,
+    ConditionOpcode.ASSERT_BEFORE_SECONDS_RELATIVE.value: AssertBeforeSecondsRelative,
+    ConditionOpcode.ASSERT_BEFORE_SECONDS_ABSOLUTE.value: AssertBeforeSecondsAbsolute,
+    ConditionOpcode.ASSERT_BEFORE_HEIGHT_RELATIVE.value: AssertBeforeHeightRelative,
+    ConditionOpcode.ASSERT_BEFORE_HEIGHT_ABSOLUTE.value: AssertBeforeHeightAbsolute,
+    ConditionOpcode.SOFTFORK.value: Softfork,
+    ConditionOpcode.REMARK.value: Remark,
+}
+
+
+CONDITION_DRIVERS_W_ABSTRACTIONS: Dict[bytes, Type[Condition]] = {
+    ConditionOpcode.AGG_SIG_PARENT.value: AggSig,
+    ConditionOpcode.AGG_SIG_PUZZLE.value: AggSig,
+    ConditionOpcode.AGG_SIG_AMOUNT.value: AggSig,
+    ConditionOpcode.AGG_SIG_PUZZLE_AMOUNT.value: AggSig,
+    ConditionOpcode.AGG_SIG_PARENT_AMOUNT.value: AggSig,
+    ConditionOpcode.AGG_SIG_PARENT_PUZZLE.value: AggSig,
+    ConditionOpcode.AGG_SIG_UNSAFE.value: AggSig,
+    ConditionOpcode.AGG_SIG_ME.value: AggSig,
+    ConditionOpcode.CREATE_COIN.value: CreateCoin,
+    ConditionOpcode.RESERVE_FEE.value: ReserveFee,
+    ConditionOpcode.CREATE_COIN_ANNOUNCEMENT.value: CreateAnnouncement,
+    ConditionOpcode.ASSERT_COIN_ANNOUNCEMENT.value: AssertAnnouncement,
+    ConditionOpcode.CREATE_PUZZLE_ANNOUNCEMENT.value: CreateAnnouncement,
+    ConditionOpcode.ASSERT_PUZZLE_ANNOUNCEMENT.value: AssertAnnouncement,
+    ConditionOpcode.ASSERT_CONCURRENT_SPEND.value: AssertConcurrentSpend,
+    ConditionOpcode.ASSERT_CONCURRENT_PUZZLE.value: AssertConcurrentPuzzle,
+    ConditionOpcode.ASSERT_MY_COIN_ID.value: AssertMyCoinID,
+    ConditionOpcode.ASSERT_MY_PARENT_ID.value: AssertMyParentID,
+    ConditionOpcode.ASSERT_MY_PUZZLEHASH.value: AssertMyPuzzleHash,
+    ConditionOpcode.ASSERT_MY_AMOUNT.value: AssertMyAmount,
+    ConditionOpcode.ASSERT_MY_BIRTH_SECONDS.value: AssertMyBirthSeconds,
+    ConditionOpcode.ASSERT_MY_BIRTH_HEIGHT.value: AssertMyBirthHeight,
+    ConditionOpcode.ASSERT_EPHEMERAL.value: AssertEphemeral,
+    ConditionOpcode.ASSERT_SECONDS_RELATIVE.value: Timelock,
+    ConditionOpcode.ASSERT_SECONDS_ABSOLUTE.value: Timelock,
+    ConditionOpcode.ASSERT_HEIGHT_RELATIVE.value: Timelock,
+    ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE.value: Timelock,
+    ConditionOpcode.ASSERT_BEFORE_SECONDS_RELATIVE.value: Timelock,
+    ConditionOpcode.ASSERT_BEFORE_SECONDS_ABSOLUTE.value: Timelock,
+    ConditionOpcode.ASSERT_BEFORE_HEIGHT_RELATIVE.value: Timelock,
+    ConditionOpcode.ASSERT_BEFORE_HEIGHT_ABSOLUTE.value: Timelock,
+    ConditionOpcode.SOFTFORK.value: Softfork,
+    ConditionOpcode.REMARK.value: Remark,
+}
+
+
+def parse_conditions_non_consensus(conditions: Iterable[Program], abstractions: bool = True) -> List[Condition]:
```

Comment:
as far as I can tell, this function is only used by tests. but maybe that's because you have a future patch in mind to use this. Is this meant to be the "entrypoint" into using these new types?

It's not trivial to undersand what `abstractions` mean here. I suspect it has to do with the extra parameters passed into the `from_program()` to augment the condition classes with extra context that's not available in the serialized condition itself.

It might warrant a comment.